newIDE/app/src/Credits/CreditsUsageDialog.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+import { Trans, t } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+import * as React from 'react';
+import { ColumnStackLayout } from '../UI/Layout';
+import FlatButton from '../UI/FlatButton';
+import Dialog from '../UI/Dialog';
+import CreditsStatusBanner from './CreditsStatusBanner';
+import RaisedButton from '../UI/RaisedButton';
+import { Column } from '../UI/Grid';
+import Text from '../UI/Text';
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import useAlertDialog from '../UI/Alert/useAlertDialog';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+
+type Props = {|
+  onClose: () => void,
+  title: React.Node,
+  message: React.Node,
+  onConfirm: () => Promise<void>,
+  successMessage: React.Node,
+|};
+
+const CreditsUsageDialog = ({
+  onClose,
+  title,
+  message,
+  onConfirm,
+  successMessage,
+}: Props) => {
+  const [isPurchasing, setIsPurchasing] = React.useState(false);
+  const [isPurchaseSuccessful, setisPurchaseSuccessful] = React.useState(false);
+  const { showAlert } = useAlertDialog();
+  const {
+    onRefreshGameTemplatePurchases,
+    onRefreshAssetPackPurchases,
+    onPurchaseSuccessful,
+    onRefreshLimits,
+  } = React.useContext(AuthenticatedUserContext);
+
+  const onPurchase = React.useCallback(
+    async () => {
+      setIsPurchasing(true);
+      try {
+        await onConfirm();
+        setisPurchaseSuccessful(true);
+        // We assume that the purchase was successful, so we refresh the purchases and limits,
+        // no need to wait for those to complete as they are not critical to the purchase.
+        Promise.all([
```

Comment:
If you don't need to await for them, you can call them directly? Or do you still need `Promise.all`?

Reply:
Good question, probably not needed indeed